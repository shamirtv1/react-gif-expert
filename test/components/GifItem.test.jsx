import {GifItem} from "../../src/components/index.js";
import {render, screen} from "@testing-library/react";

describe("Pruebas en GifItem", function() {

    const title = "Titulo del gif"
    const url = "http.url.del.gif"

    test("Evaluar el snapshot", function() {
        const { container } = render(<GifItem title={title} url={url} />);
        expect( container ).toMatchSnapshot();
    });

    test("Debe de mostrar la imagen con el URL y el ALT indicado", function() {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(src);
        expect( alt ).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', ()=>{
        render(<GifItem title={title} url={url} />);
        expect(screen.getByRole('paragraph').innerHTML).toBe(title);
    });

})