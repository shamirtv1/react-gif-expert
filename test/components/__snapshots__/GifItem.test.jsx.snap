// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Evaluar el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo del gif"
      src="http.url.del.gif"
    />
    <p>
      Titulo del gif
    </p>
  </div>
</div>
`;
